xds/internal/xdsclient/tests/resource_update_test.go

```diff
@@ -561,8 +561,8 @@ func (s) TestHandleRouteConfigResponseFromManagementServer(t *testing.T) {
 			bc := e2e.DefaultBootstrapContents(t, nodeID, mgmtServer.Address)
 			client, close, err := xdsclient.NewForTesting(xdsclient.OptionsForTesting{
 				Name:               t.Name(),
-				Contents:           bc,
 				WatchExpiryTimeout: defaultTestWatchExpiryTimeout,
+				Contents:           bc,
```

Comment:
These tests should be using a pool to create an xDS client in. In fact, once we delete these `NewXxx` functions which don't operate on a pool, the tests will be left with no other choice.